www_src/pages/tinker/tinker.jsx

```diff
@@ -1,68 +1,157 @@
 var React = require('react');
-
 var render = require('../../lib/render.jsx');
+var api = require('../../lib/api');
+var elementTypes = require('../../components/el/el.jsx').types;
+var assign = require('react/lib/Object.assign');
 
 var Tabs = require('../../components/tabs/tabs.jsx');
 var ColorSpectrum = require('../../components/color-spectrum/color-spectrum.jsx');
-var Slider = require('../../components/range/range.jsx');
+var Loading = require('../../components/loading/loading.jsx');
+
+/**
+ * Tinker: Advanced editor for specific element attributes or styles
+ *
+ * Props
+ *   @key {boolean} isVisible Indicates if the Tinker Activity is currently visible (not paused)
+
+ * State
+ *   @key {boolean} loading If true, loading spinner is shown and UI is blocked
+ *   @key {object} element Represents flattened element data (via spec.flatten)
 
+ *   From router:
+ *   @key {object} params Contains params user, project, page, element, propertyName
+ */
 var Tinker = React.createClass({
+
+  mixins: [require('../../lib/router')],
+
   getInitialState: function () {
     return {
-      color: '#8FB70A'
+      loading: false,
+      element: null
     };
   },
-  onChangeColor: function (color) {
+
+  componentDidMount: function () {
+
+    // Prevent pull to refresh
+    document.body.style.overflowY = 'hidden';
```

Comment:
can we take out this hack and make that a class that we can use instead? If it's style, let's not JS hardcode it - we'll never find it again =(